src/parser/core/modules/NotCasting.tsx

```diff
@@ -0,0 +1,184 @@
+import {t} from '@lingui/macro'
+import {Trans} from '@lingui/react'
+import {Action} from 'data/ACTIONS'
+import {Event, Events} from 'event'
+import {Analyser} from 'parser/core/Analyser'
+import {filter} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import Downtime from 'parser/core/modules/Downtime'
+import {GlobalCooldown} from 'parser/core/modules/GlobalCooldown'
+import {Timeline} from 'parser/core/modules/Timeline'
+import React from 'react'
+import {Table, Button} from 'semantic-ui-react'
+import {Data} from './Data'
+
+//value to be added to the gcd to avoid false positives. 100ms for caster tax, 50ms for gcd jitter.
+const GCD_ERROR_OFFSET = 150
+const BASE_RECAST = 2500
+
+interface Window {
+	start: number,
+	stop?: number,
+	recastTime: number,
+}
+
+export class NotCasting extends Analyser {
+	static override handle = 'notcasting'
+	static override title = t('core.notcasting.title')`Times you did literally nothing`
+
+	@dependency private timeline!: Timeline
+	@dependency protected gcd!: GlobalCooldown
+	@dependency protected downtime!: Downtime
+	@dependency protected data!: Data
+
+	// give some options depending on what is optimal for that job
+	protected includeOGCDs: boolean = false
+
+	private noCastWindows: {current?: Window, history: Window[]} = {
+		history: [],
+	}
+	private hardCastStartTime: number | undefined = undefined
+
+	override initialise() {
+		const playerFilter = filter<Event>().source(this.parser.actor.id)
+		this.addEventHook(playerFilter.type('prepare'), this.onBegin)
+		this.addEventHook(playerFilter.type('action'), this.onCast)
+		this.addEventHook({
+			type: 'death',
+			actor: this.parser.actor.id,
+		}, this.onDeath)
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private onCast(event: Events['action']) {
+		//some jobs might care about oGCDs. e.g. BLM
+		const actionIsOGCD: boolean = !this.data.getAction(event.action)?.onGcd ?? false
```

Comment:
This line's giving me a "right operand of ?? is unreachable" error. Either isn't necessary, or need to rewrite it to properly trap the null